### PR TITLE
feat: e-sport mode

### PR DESCRIPTION
**Pull request by OxrxL**
https://github.com/revanced/revanced-integrations/pull/146

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        )
+
+        val timeCounterMethod = TimeCounterFingerprint.result!!.mutableMethod
+
+        timeCounterMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/EsportModePatch;->esportModeEnabled()Z
+            move-result v0
+            if-eqz v0, :esport_mode_enabled
+            const-string v0, "xx:xx:xx"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt:63]:
```suggestion
            const-string v0, ""
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
While it's a fun name, to be consistent and clear, use a different name for the patch such as "hide-time-and-seekbar".

File: src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt (hunk 1 above)
Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt:63]: A user explictly request me to add this as time counter, instead nothing.

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt:63]: Other users will request to remove. if you want to keep it, make it toggleable on-click.

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidetimeandseekbar/patch/HideTimeAndSeekbarPatch.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                StringResource("revanced_hide_time_and_seekbar_summary_on", "Time and seekbar are hidden"),
+                StringResource("revanced_hide_time_and_seekbar_summary_off", "Time and seekbar are visible")
+            )
+        )
+
+        val inlineTimeBarWrapperMethod = InlineTimeBarWrapperFingerprint.result!!.mutableMethod
+
+        inlineTimeBarWrapperMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/HideTimeAndSeekbarPatch;->hideTimeAndSeekbar()Z
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidetimeandseekbar/patch/HideTimeAndSeekbarPatch.kt:47]: The integrations code for this is missing, please do not forget PRing this before marking this PR as ready.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidetimeandseekbar/patch/HideTimeAndSeekbarPatch.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import app.revanced.patches.youtube.layout.hidetimeandseekbar.fingerprints.TimeCounterFingerprint
+import app.revanced.patches.youtube.layout.hidetimeandseekbar.annotations.HideTimeAndSeekbarCompatibility
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("e-sport-mode")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidetimeandseekbar/patch/HideTimeAndSeekbarPatch.kt:23]:
```suggestion
@Name("hide-time-and-seekbar")
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by OxrxL** [comment]
First comment updated with integrations.

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**